File: src/solutions/004/004_arvindch.py
#!/usr/bin/env python2.7

__author__ = 'Arvind Chembarpu'
__email__ = 'achembarpu@gmail.com'


from src.custom import checks


def main():
    
    giv_low = 100
    giv_high = 1000
    req_num = 0
    
    for a in xrange(giv_low, giv_high):
        for b in xrange(a, giv_high):
            product = a * b
            if checks.is_palindrome(product) and product > req_num:
                req_num = product
    
    return req_num


if __name__ == '__main__':
    answer = main()
    print answer

File: src/java/solutions/004/_004_arvindch.java
public class _004_arvindch {

    public static void main(String[] args) {
        int p=0, giv_low=100, giv_high=1000, req_num=0;

        for(int a=giv_low; a<giv_high; a++) {
            for(int b=a; b<giv_high; b++) {

                p = a * b;
                if(p>req_num) {
                    String s = Integer.toString(p);
                    String rs = new StringBuffer(s).reverse().toString();

                    if(s.equals(rs))
                        req_num = p;
                }
            }
        }

        System.out.println(req_num);
    }

}
